### DISCUSSION TITLE
Different object amount in same layer using FreeCAD and LibreDWG

### DISCUSSION THREAD

**Discussion by giedrius-N** [LibreDWG/libredwg#1027] [Q&A]
Hello,

I encountered an issue where the number of objects read from a DWG file using LibreDWG does not match the number of objects visible in FreeCAD. Specifically, when I open the same DWG file in FreeCAD, I get fewer objects than what LibreDWG reads in that particular layer. My application reads extra objects, which I suspect are unwanted lines or polylines.

- In FreeCAD, I see only 14 objects in a particular layer, which is correct.
- When I use LibreDWG to parse the same file, I get 26 objects in the same layer, which includes 12 extra simple lines and polylines that should not exist.

These extra objects appear in specific layers and show up **in the left corner of the drawing**. This behavior is incorrect because they are not present in FreeCAD, and they should not be there according to the original file.

Things I’ve tried:
- I’ve confirmed that all layers are visible in FreeCAD, and nothing is hidden. So, the extra objects are definitely not on hidden layers.
- I checked for blocks or nested elements in the DWG file that could contain these extra lines/polylines, but nothing seems out of place in FreeCAD.
- I attempted to clean up the file and remove any redundant objects, but this did not change the outcome when opening the file in LibreDWG.

**Questions**

- Why might LibreDWG be interpreting these extra lines and polylines, while FreeCAD does not show them?
- How can I reliably filter out or ignore these unwanted objects in my program? Are there any specific flags or properties in DWG objects that LibreDWG is reading, which FreeCAD might be ignoring?
- Could this be related to block definitions or some kind of auxiliary geometry that FreeCAD hides by default? If so, how should I handle these cases programmatically?

![image](https://github.com/user-attachments/assets/adb84777-6333-410f-a8b9-6dce7a5bfe95)
![image](https://github.com/user-attachments/assets/3855573f-bfbf-42a2-bd32-d77bfad06407)

Those above in red rectangle are unwanted objects.
In second image there is what I see in FreeCAD

Any help or guidance would be greatly appreciated! Thanks in advance!

**Comment by michal-josef-spacek** [comment]
As I understand, FreeCAD uses different approaches to read DWG files (libredwg dwg2dxf or ODA converter or qcad) to convert to DXF.

Some hints:
1) Test on DXF file.
2) Check which one of the converters is used.
